[Feature] Introduce per-instance sampling APIs for long task metric.

jianliang00 opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lynx-family/lynx?pullRequest=320) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/lynx-family/lynx?pullRequest=320) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lynx-family/lynx?pullRequest=320) it.</sub>

kazec committed:
[Feature] Introduce per-instance sampling APIs for long task metric.

This MR Introduced a new common struct InstanceOptions to encapsulate common
params used by components within a lynx template assembler instance.

Additionally this MR introduced per LynxView APIs for long task metric configuration,
and refactored long task monitoring to use information from InstanceOptions to determine
whether the monitoring should be enabled for a long task.

Major changes includes:
1. Add LynxView.setLongTaskMonitorEnabled APIs in the iOS/Android code.
2. Introduce new InstanceOptions in C++ code to encapsulate common
params used by components within a lynx template assembler instance.
3. Passing longTaskMonitorEnabled flag down to various C++ components using InstanceOptions: shell, engine, runtime, jsi, radon, etc.
4. Update call-sites of LongTaskMonitor to accept InstanceOptions.
5. Update the interface and implementation of LongTaskMonitor to accept InstanceOptions.

doc: https://github.com/lynx-family/lynx/issues/552
